dev-docs/RFCs/v6.0/offscreen-render-rfc.md

```diff
@@ -0,0 +1,100 @@
+
+Static methods:
+- `createWorker(opts)` - returns script that can be used to spin up a new Worker. `opts` are the options used to construct an `AnimationLoop` instance on the worker thread.
+
+Constructor:
+- `new OffscreenAnimationLoop(opts)` - `opts` are the same options used to construct an `AnimationLoop`, with additional fields:
+    + `opts.worker` (Worker) - the worker instance running a script created by `OffscreenAnimationLoop.createWorker`
+
+Methods:
+- `setProps(props)` - equivalent to `animationLoop.setProps(props)`
+- `start(opts)` - equivalent to `animationLoop.start(opts)`
+- `stop()` - equivalent to `animationLoop.stop()`
+
+
+### Examples
+
+```
+/// animation.js
+import {GL, Model, Framebuffer, Matrix4} from 'luma.gl';
+
+export default {
+    onInitialize: () => {
+        // set up scene
+    },
+    onRender: () => {
+        // render frame
+    },
+    glOptions: {},
+    autoResizeDrawingBuffer: true
+};
+```
+
+Render on main thread:
+```
+/// app.js
+import animationLoopOptions from './animation';
+import {AnimationLoop} from 'luma.gl';
+
+const animationLoop = new AnimationLoop(animationLoopOptions);
+animationLoop.start({canvas: 'lumagl-canvas'});
+```
+
+Render on worker thread:
+```
+/// app.js
+import animationLoopOptions from './animation';
+import {OffscreenAnimationLoop} from 'luma.gl';
+import webworkify from 'webworkify';
+
+const worker = OffscreenAnimationLoop.createWorker(animationLoopOptions);
+const animationLoop = new OffscreenAnimationLoop({
+    ...animationLoopOptions,
+    worker: webworkify(worker)
+});
+animationLoop.start({canvas: 'lumagl-canvas'});
+```
+
+
+## Proposal: Interactivity
+
+`AnimationLoop` does not directly handle user input (mousemove, click, etc.). This is usually not an issue as the user can update their own application state from event callbacks and then access that state from within `onRender`. In the offscreen rendering case, this is no longer possible as `onRender` is on a different thread from where the event callbacks are executed.
+
+The proposed solution is to expose a new API on both `AnimationLoop` and `OffscreenAnimationLoop` for applications to send custom data to the renderer:
+
+- Add `setUserData` method to `AnimationLoop`. It adds a field `userData` to `animationLoop._callbackData` which can be accessed in the `onRender` callback.
+- Add `setUserData` method to `OffscreenAnimationLoop`. It posts the user data to the worker which then calls `animationLoop.setUserData`.
+
+
+## Challenges
+
+The main challenge in this effort is that DOM properties are accessed in various places of the code base (e.g. webgl-utils, AnimationLoop, hover/click handling, viewport controls). Certain globals such as `window` `document` and `HTMLCanvasElement` are not available to the worker thread. We should either improve the compatibility of existing code, or invest in separating DOM-dependent code from the core.
```

Comment:
This has been resolved, right? Maybe change the tense of the sentence to "maintaining" or similar.

Reply:
Not entirely. We still have these scattered in the code base, e.g. picking.